sdk/lib/drivers/lwip/src/core/dhcp.c

```diff
@@ -1377,7 +1377,11 @@ dhcp_parse_reply(struct dhcp *dhcp, struct pbuf *p)
   offset_max = options_idx_max;
   options = (u8_t*)q->payload;
   /* at least 1 byte to read and no end marker, then at least 3 bytes to read? */
+#ifdef __REACTOS__
+  while ((q != NULL) && (offset < offset_max) && (options[offset] != DHCP_OPTION_END)) {
```

Comment:
If this is fixed upstream (i.e. this is a partial sync), you don't need the ifdef

Reply:
`#ifdef` removed.